Data/Primitive/Array/Class.hs

```diff
@@ -140,8 +140,10 @@ class Arr (marr :: * -> * -> *) (arr :: * -> * ) a | arr -> marr, marr -> arr wh
     -- | Resize a mutable array to the given size.
     resizeMutableArr :: (PrimMonad m, PrimState m ~ s) => marr s a -> Int -> m (marr s a)
 
-    -- | Shrink a mutable array to the given size. This operation only works on primitive arrays.
-    -- For boxed arrays, this is a no-op, e.g. 'sizeOfMutableArr' will not change.
+    -- | Shrink a mutable array to the given size.
+    --
+    -- This operation is not guaranteed to have effects, e.g. 'sizeOfMutableArr' will not change.
```

Comment:
i think you could say  something about "this operation is instance dependent, but behavior could be ..."